cache/refs.go

```diff
@@ -848,6 +848,11 @@ func getBlobWithCompression(ctx context.Context, cs content.Store, desc ocispecs
 }
 
 func walkBlob(ctx context.Context, cs content.Store, desc ocispecs.Descriptor, f func(ocispecs.Descriptor) bool) error {
+	if _, err := cs.Info(ctx, desc.Digest); errors.Is(err, cerrdefs.ErrNotFound) {
```

Comment:
Do we know a more defined call path for this case? I assume the blob is missing because it was not pulled and is missing the `descHandlers`. Is this expected or did we forget to pass the `descHandlers`? Or is this a callpath where we would want to skip a blob if we only have access to it remotely and not locally?

Reply:
This is called from `getAvailableBlobs` and in this context, this function doesn't have access to refs(and descHandlers) to compression variants thus GetRemtoes currently treats only local blobs as available compression variants. To change GetRemote to return all remotely accessible variants, we need to change `getAvailableBlobs` somehow accessible to all compression variant's refs and descHandlers to trigger unlazying of them.